### PR TITLE
Messaging improvements

### PR DESCRIPTION
**Pull request by BasBuijsen**
This pr adds some functionality for the messaging framework that is most important in iOS and Android projects. It is only (un)subscribing to topics and retrieving FCM token. Adding more functionality is going to be hard because of the differences between the sdks.

This pr also has 2 very small improvements from my previous PR's.
- Add analytics to readme
- Fix file names in tests for storage to ensure parallelization is not a problem

### CODE REVIEW

**Comment by BasBuijsen** [comment]
@nbransby can you review this pr?

File: firebase-messaging/src/jsMain/kotlin/dev/gitlive/firebase/messaging/messaging.kt [hunk 1]
```diff
@@ -3,10 +3,22 @@ package dev.gitlive.firebase.messaging
...
 actual class FirebaseMessaging(val js: Messaging) {
+    actual fun subscribeToTopic(topic: String) {
+        // This is not supported in the JS SDK
+        // https://firebase.google.com/docs/reference/js/messaging_.md#@firebase/messaging
+    }
+
+    actual fun unsubscribeFromTopic(topic: String) {
+        // This is not supported in the JS SDK
+        // https://firebase.google.com/docs/reference/js/messaging_.md#@firebase/messaging
+    }
```
[the remark this answers is not in the document]
Comment (BasBuijsen) [reply firebase-messaging/src/jsMain/kotlin/dev/gitlive/firebase/messaging/messaging.kt:21]: Yea we should. I will add it.

File: firebase-messaging/src/jsMain/kotlin/dev/gitlive/firebase/messaging/externals/messaging.kt [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 ): Messaging
 
-external interface Messaging
\ No newline at end of file
+external fun getToken(messaging: Messaging = definedExternally, options: dynamic = definedExternally): Promise<String>
+
+external interface Messaging {
+    fun subscribeToTopic(tokens: Array<String>, topic: String): Promise<Unit>
+    fun unsubscribeFromTopic(tokens: Array<String>, topic: String): Promise<Unit>
+}
```
Comment (nbransby) [inline firebase-messaging/src/jsMain/kotlin/dev/gitlive/firebase/messaging/externals/messaging.kt:15]: remove this since it doesn't exist?

**Comment by BasBuijsen** [comment]
@nbransby it seems that the android tests keep running infinitely while there were already errors. Could you stop it before it keeps running for hours :)

File: firebase-messaging/src/jsMain/kotlin/dev/gitlive/firebase/messaging/externals/messaging.kt (hunk 2 above)
Comment (BasBuijsen) [reply firebase-messaging/src/jsMain/kotlin/dev/gitlive/firebase/messaging/externals/messaging.kt:15]: I will

**Comment by Reedyuk** [comment]
> @nbransby it seems that the android tests keep running infinitely while there were already errors. Could you stop it before it keeps running for hours :)

rebase now and see if it works

**Comment by BasBuijsen** [comment]
Yay tests work now 🎉

**Comment by Reedyuk** [comment]
> Yay tests work now 🎉

rocky-rockyiv-apollo-creed-stallone-carlweathers-bromance-bros-beach-fun-training-guys-americans-positivity-boxing-brosbeforehos (1)

**Pull request merged**